File: nodejs/src/factory/VietnameseDrinkFactory.ts
import DrinkFactory from "./DrinkFactory";
import VietnameseCoffee from "./products/VietnameseCoffee";
import VietnameseTea from "./products/VietnameseTes";

class VietnameseDrinkFactory extends DrinkFactory {
    createCoffee() {
        return new VietnameseCoffee();
    } // Factory Method
    createTea() {
        return new VietnameseTea();
    } // Factory Method
}

export default VietnameseDrinkFactory;

File: spring/demo/src/main/java/com/example/demo/factory/VietnameseDrinkFactory.java
package com.example.demo.factory;

import com.example.demo.factory.products.*;

public class VietnameseDrinkFactory extends DrinkFactory {
    @Override public Coffee createCoffee() { return new VietnameseCoffee(); }
    @Override public Tea createTea() { return new VietnameseTea(); }
}
